tests/component/_task_modules/test_triggers_properties.py

```diff
@@ -99,3 +109,45 @@ def test___ai_task___reset_uint32_property___returns_default_value(ai_voltage_ta
     del ai_voltage_task.triggers.reference_trigger.pretrig_samples
 
     assert ai_voltage_task.triggers.reference_trigger.pretrig_samples == 2
+
+
+def test___ai_voltage_field_daq_task___get_timestamp_property___returns_default_value(
+    ai_voltage_field_daq_task: Task,
+):
+    ai_voltage_field_daq_task.timing.cfg_samp_clk_timing(1000)
+
+    when_value = ai_voltage_field_daq_task.triggers.start_trigger.trig_when
+
+    assert when_value.lsb == 0
+    assert when_value.msb == 0
+
+
+def test___ai_voltage_field_daq_task___set_timestamp_property___returns_assigned_value(
+    ai_voltage_field_daq_task: Task,
+):
+    value_to_test = JAN_01_2002_HIGHTIME
+    ai_voltage_field_daq_task.timing.cfg_samp_clk_timing(1000)
+
+    ai_voltage_field_daq_task.triggers.start_trigger.trig_when = value_to_test
+
+    when_value = ai_voltage_field_daq_task.triggers.start_trigger.trig_when
+    when_value_dt = when_value.to_datetime(timezone.utc)
+    assert when_value_dt.year == 2002
+    assert when_value_dt.month == 1
+    assert when_value_dt.day == 1
+    assert when_value_dt.hour == 0
+    assert when_value_dt.minute == 0
+    assert when_value_dt.second == 0
+
+
+def test___ai_voltage_field_daq_task___reset_timestamp_property___returns_default_value(
+    ai_voltage_field_daq_task: Task,
+):
+    ai_voltage_field_daq_task.timing.cfg_samp_clk_timing(1000)
+    ai_voltage_field_daq_task.triggers.start_trigger.trig_when = JAN_01_2002_HIGHTIME
+
+    del ai_voltage_field_daq_task.triggers.start_trigger.trig_when
+
+    when_value = ai_voltage_field_daq_task.triggers.start_trigger.trig_when
+    assert when_value.lsb == 0
+    assert when_value.msb == 0
```

Comment:
Wait a minute, `hightime` doesn't have `lsb` and `msb` fields. This property is returning the wrong data type.

`set_trig_attribute_timestamp` is correctly converting from `hightime`/`datetime` to `AbsoluteTime`: https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_library_interpreter.py#L5350

`get_trig_attribute_timestamp` should convert from `AbsoluteTime` back to `hightime` before returning, but it's not doing this: https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_library_interpreter.py#L3674

Reply:
You are right, I have fixed this in the next commit. 

However, this introduced a whole new level of complexity. According to the logic in `AbsoluteTime`, the default value is lsb = 0 and msb = 0, which means it should be 1904-01-01. However, I am getting OS error when trying to use `fromtimestamp()` with the `_BIAS_FROM_1970_EPOCH` which is -2082844800. [According to the documentation ](https://docs.python.org/3/library/datetime.html#:~:text=This%20may%20raise%20OverflowError,ValueError%20on%20localtime()%20failure), it seems like the number is out of scope of the supported range. [Same goes to astimezone()](https://bugs.python.org/issue36759#:~:text=This%20exception%20is,raise%20a%20ValueError%3F). 

So I implemented a small change in _lib_time which helps us address this small corner case. 

Might be good to note that `fromtimestamp()` does pass with -2082844800 when I run it on an online python compiler, so this leads me to believe it could be a python issue. I hit OS error also when I try to run it on Python 3.6.

Reply:
Argh, that's unfortunate. I don't like your workaround in `AbsoluteTime`. It is a hack that only works for time 0, which is isn't a relevant use-case. It is unfortunate that simulated devices have this behavior, but I don't think we should work around it. You could either make the test expect the exception or use a real device or both. I'm leaning towards both. For the real-device test:
* Getting a time-aware device is not always easy
* I'm not sure we'd want to force the driver ownership team to validate that on release every time
* **but** writing the test and performing a one-time functional validation with real hardware sounds like a great idea. I think we'll run into more of these, so I'm fine delaying that final validation with real hardware until all the time stuff is in.

Reply:
There is a simpler way to handle dates before 1970:
- Construct a `datetime` set to the 1904-01-01 epoch
- Add `timedelta(seconds=msb)` to it
- Add `timedelta(seconds=lsb/(2**64))` to it

See https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/daqmx/api/nidaqmxSharp/source/Api/DaqTypeConversion.cs

Reply:
@zhindes 

> It is unfortunate that simulated devices have this behavior, but I don't think we should work around it.

It seems that even physical devices's default value for Start Time When is 0 Absolute Time.

Default value on physical 9189
![image](https://github.com/ni/nidaqmx-python/assets/74756143/cf3dd73d-8bc8-4bcd-9296-3088d03693a0)

Default value on simulated fieddaq
![image](https://github.com/ni/nidaqmx-python/assets/74756143/edddcaf6-9a55-441c-9653-fa87fc7edf32)

Reply:
@DeborahOoi96 Those screenshots aren't doing what you think. They are returning the default value of the Start.Time.When attribute, which is 0. That's a reasonable default and will make your `reset` test a bit of a challenge. However, if you set that to a non-default value, it should maintain it unless there is some weird driver behavior doing something different.

> based on Zach's comment I think he suggests moving towards not having this kind of hack in libtime since it is not a relevant use case.

I definitely don't want a hack in libtime, but what Brad described doesn't seem like a hack to me.

```python
>>> time0 = datetime.datetime(1904,1,1)
>>> td = datetime.timedelta(seconds=10)
>>> print(time0)
1904-01-01 00:00:00
>>> print(td)
0:00:10
>>> time0 + td
datetime.datetime(1904, 1, 1, 0, 0, 10)
>>> time0 - td
datetime.datetime(1903, 12, 31, 23, 59, 50)
```

Reply:
To be more clear, handling `time==0` specifically as some special case is not great. Having some generic code that handles times before the unix 1970 epox is fine.

Reply:
Deleting an attribute in nidaqmx-python is equivalent to resetting an attribute to default value in LabVIEW.
![image](https://github.com/ni/nidaqmx-python/assets/17914399/56021fc6-c6d8-4b15-b476-caeb19b49479)
After you select "Default Value", the wire terminal for the property value goes away:
![image](https://github.com/ni/nidaqmx-python/assets/17914399/fa0eb5ff-fdfb-4ff8-8cfe-b47c3bbbaad1)